### ISSUE TITLE
How to re-import all the docs (with specific fields) again into solr from mongodb

### ISSUE THREAD

**Issue by akkina9** [yougov/mongo-connector#165]
Hello,

I just started playing with mongo connector on windows. By default, when I ran it it loaded all the docs in all the databases in mongo into solr. But I want is a particular collection and only a certain fields. So I did the following:
1) Removed all the docs in Solr by doing:
- localhost:8983/solr/update?stream.body=<delete><query>_:_</query></delete>
- localhost:8983/solr/update?stream.body=<commit/>

2) Stopped Solr, mongodb and mongo connector

3) Deleted the files in the data/index and data/tlog of Solr

4) Delete config.txt in Scripts folder for mongo-connector

5) Restarted Solr and mongodb
6) Ran the following mongo connector command to only import from one collection and certain fields.
C:\Python34\Scripts>mongo-connector -m localhost:27017 -t http://localhost:8983/solr -d c:\Python34\Lib\site-packages\mongo_connector\doc_managers\solr_doc_manager.py --auto-commit-interval=0 -n userdb.users --unique-key=id -i name -v

I do not see any docs in solr.

Here is the what shows up in the console.
2014-09-17 11:49:08,423 - INFO - Beginning Mongo Connector
2014-09-17 11:49:08,628 - DEBUG - Starting request to 'http://localhost:8983/solr/admin/luke
with body ''...
2014-09-17 11:49:08,649 - INFO - Starting new HTTP connection (1): localhost
2014-09-17 11:49:08,686 - DEBUG - "GET /solr/admin/luke?show=schema&wt=json HTTP/1.1" 200 No
2014-09-17 11:49:08,705 - INFO - Finished 'http://localhost:8983/solr/admin/luke?show=schema
 in 0.076 seconds.
2014-09-17 11:49:08,724 - INFO - MongoConnector: Empty oplog progress file.
2014-09-17 11:49:08,733 - INFO - OplogThread: Initializing oplog thread
2014-09-17 11:49:08,740 - INFO - MongoConnector: Starting connection thread MongoClient('loc
2014-09-17 11:49:08,741 - DEBUG - OplogThread: Run thread started
2014-09-17 11:49:08,742 - DEBUG - OplogThread: Getting cursor
2014-09-17 11:49:08,743 - DEBUG - OplogThread: reading last checkpoint as None
2014-09-17 11:49:08,743 - DEBUG - OplogThread: Dumping set of collections ['userdb.users']
2014-09-17 11:49:09,026 - DEBUG - OplogThread: Last entry is the one we already processed.
2014-09-17 11:49:10,026 - DEBUG - OplogThread: Getting cursor
2014-09-17 11:49:10,027 - DEBUG - OplogThread: reading last checkpoint as None
2014-09-17 11:49:10,028 - DEBUG - OplogThread: Dumping set of collections ['userdb.users']
2014-09-17 11:49:10,309 - DEBUG - OplogThread: Last entry is the one we already processed.

**Comment by dandye** [comment]
I ran into this today and found this issue because of the `AttributeError: 'NoneType' object has no attribute 'time'` line.  

My problem was due to an empty mongo collection, which caused the creation of a zero byte oplog.timestamp file (when supplying the `--no-dump` flag).  

I found I could work around the issue by either:
 1. adding data to the collection before starting mongo_connector 
 2. dropping the `--no-dump` flag when starting mongo_connector
 3. pre-creating an `oplog.timestamp` file with valid data before starting mongo_connector
